### PR TITLE
check nil pointer and throw panic

### PR DESCRIPTION
**Pull request by SinghamXiao**
Signed-off-by: yuanxiao <yuan.xiao5@zte.com.cn>

### CODE REVIEW

**Comment by SinghamXiao** [comment]
@dmcgowan Sorry for some error operations. I add a new one.

File: registry/handlers/app.go [hunk 1]
```diff
@@ -416,7 +416,11 @@ func (app *App) register(routeName string, dispatch dispatchFunc) {
 	// replace it with manual routing and structure-based dispatch for better
 	// control over the request execution.
 
-	app.router.GetRoute(routeName).Handler(app.dispatcher(dispatch))
+	route := app.router.GetRoute(routeName)
+	if route == nil {
+		panic(fmt.Sprintf("%s is not definded in routeDescriptors", routeName))
```
Comment (dmcgowan) [inline registry/handlers/app.go]: `definded` -> `defined`

Comment (SinghamXiao) [inline registry/handlers/app.go]: thx

**Comment by stevvooe** [comment]
Why? Won't it just throw a panic when trying to call the function?

In general, this shouldn't happen. What panic is trying to be fixed here?

Let's address root causes, not symptoms.

**Comment by dmcgowan** [comment]
I am going to close this PR. I understand the change is trying to provide a more meaningful message when this panic occurs, but since this is a non-functional change and the original panic can be traced to the point of failure from the nil pointer error, this change is not necessary.

**Pull request closed without merging**